lib/logstash/outputs/elasticsearch/buffer.rb

```diff
@@ -0,0 +1,130 @@
+require 'concurrent'
+java_import java.util.concurrent.locks.ReentrantLock
+
+module LogStash; module Outputs; class ElasticSearch
+  class Buffer
+    def initialize(logger, max_size, flush_interval, &block)
```

Comment:
Yes, the idea was to get the retry logic out of the buffer impl and
encapsulate the whole thing in a class rather than a mixin. My goal was to
move us incrementally forward for the next gen pipeline, which will be more
similar to this class.

On Wednesday, November 4, 2015, Jordan Sissel <notifications@github.com>
wrote:

> In lib/logstash/outputs/elasticsearch/buffer.rb
> <https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/292#discussion_r43966962>
> :
>
> > @@ -0,0 +1,130 @@
> > +require 'concurrent'
> > +java_import java.util.concurrent.locks.ReentrantLock
> > +
> > +module LogStash; module Outputs; class ElasticSearch
> > +  class Buffer
> > +    def initialize(logger, max_size, flush_interval, &block)
>
> Conceptually, this class replaces Stud::Buffer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/292/files#r43966962>
> .
>